src/az_ulib_ipc/az_ulib_ipc.c

```diff
@@ -734,101 +732,85 @@ AZ_NODISCARD az_result az_ulib_ipc_split_method_name(
   AZ_ULIB_TRY
   {
     uint32_t package_version_temp;
-    int32_t interface_name_field;
     uint32_t interface_version_temp;
-    bool has_capability_name;
-    int32_t count_field = 0;
-    az_span fields[MAX_FIELDS_IN_METHOD_FULL_NAME];
+    az_span device_name_temp;
+    az_span package_name_temp;
+    az_span interface_name_temp;
+    az_span capability_name_temp;
     int32_t split_pos;
 
-    do // Split the entire method full name using `.` as marker.
+    // Extract the device name, if provided.
+    split_pos = az_span_find(full_name, AZ_SPAN_FROM_STR("@"));
+    if (split_pos > 0)
     {
-      split_pos = az_span_find(full_name, AZ_SPAN_FROM_STR("."));
-      if (split_pos > 0)
-      {
-        fields[count_field] = az_span_slice(full_name, 0, split_pos);
-        full_name = az_span_slice_to_end(full_name, split_pos + 1);
-        count_field++;
-      }
-    } while ((split_pos > 0) && (count_field < MAX_FIELDS_IN_METHOD_FULL_NAME));
-    count_field--;
-
-    // Last field of full_name is the capability_name or interface_version and shall not be
-    // AZ_SPAN_EMPTY.
-    AZ_ULIB_THROW_IF_ERROR((az_span_size(full_name) > 0), AZ_ERROR_UNEXPECTED_CHAR);
-
-    // interface_name shall not be AZ_SPAN_EMPTY.
-    AZ_ULIB_THROW_IF_ERROR((count_field >= 0), AZ_ERROR_UNEXPECTED_CHAR);
```

Comment:
are you still checking that capability is not empty or does that happen with the ipc call?

Reply:
Capability and device name may be empty. The user of this API shall decide if it is mandatory or not.